### PR TITLE
fix multibyte utf-8 conversion of escapeurl 

### PR DESCRIPTION
**Pull request by HereThereBeDragons**
Issue #3368

as in `download.cc` the functions are `static` i know copied them to `t_download.cc` and added a comment to them.

if another solution is preferred, pls let me know

### CODE REVIEW

**Comment by HereThereBeDragons** [comment]
@cernvm-bot cloudtest

**Comment by HereThereBeDragons** [comment]
@cernvm-bot cloudtest

File: cvmfs/network/download.cc [hunk 1]
```diff
@@ -416,6 +336,84 @@ const int DownloadManager::kProbeUnprobed = -1;
...
+    }
+  }
+  LogCvmfs(kLogDownload, kLogDebug, "escaped %s to %s",
+           url.c_str(), escaped.c_str());
+
+  return escaped;
+}
+
+/**
+ * escaped array needs to be sufficiently large.  It's size is calculated by
```
Comment (vvolkl) [inline cvmfs/network/download.cc]:
```suggestion
 * escaped array needs to be sufficiently large.  Its size is calculated by
```
typo

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest

**Pull request merged**